### ISSUE TITLE
Opencv features linked, though default-features is false and features not included

### ISSUE THREAD

**Issue by buddhisthead** [twistedfall/opencv-rust#517]
Summary:
```
/usr/bin/ld: cannot find -lopencv_contrib
/usr/bin/ld: cannot find -lopencv_legacy
```

While running cargo build, all goes well until the linker phase, where these libs are not found. They are not installed on my system, however I've specified: `opencv = { version = "0.86.1", default-features = false, features = ["imgcodecs", "imgproc", "clang-runtime"] }` in the Cargo.toml file in an attempt to not require them. We build for a headless system and have reduced libraries in the runtime enviroment (Jetson Xavier/Orin). I'm unclear on why these features are required after specifying no defaults.

Any suggestions for further data that would be helpful?

1. Operating system: Ubuntu 20.04
2. OpenCV Built from source, but without `contrib` or `legacy`
4. OpenCV version: 4.5.5
6. rustc version: rustc 1.72.0 (5680fa18f 2023-08-23)
8. Attach the full output of the following command from your project directory:
   ```shell script
   RUST_BACKTRACE=full cargo build -vv 
       Fresh libc v0.2.150
       Fresh memchr v2.6.4
       Fresh cfg-if v1.0.0
       Fresh autocfg v1.1.0
       Fresh glob v0.3.1
       Fresh unicode-ident v1.0.12
       Fresh aho-corasick v1.1.2
       Fresh libloading v0.7.4
       Fresh jobserver v0.1.26
       Fresh dunce v1.0.4
       Fresh proc-macro2 v1.0.69
       Fresh futures-core v0.3.29
       Fresh bitflags v2.4.1
       Fresh once_cell v1.18.0
       Fresh futures-sink v0.3.29
       Fresh linux-raw-sys v0.4.11
       Fresh percent-encoding v2.3.0
       Fresh clang-sys v1.6.1
       Fresh quote v1.0.33
       Fresh futures-channel v0.3.29
       Fresh rustix v0.38.25
       Fresh slab v0.4.9
       Fresh futures-task v0.3.29
       Fresh cc v1.0.79
       Fresh semver v1.0.20
       Fresh clang v2.0.0
       Fresh syn v2.0.39
       Fresh vcpkg v0.2.15
       Fresh futures-io v0.3.29
       Fresh pin-utils v0.1.0
       Fresh log v0.4.20
       Fresh regex-syntax v0.8.2
       Fresh shlex v1.2.0
       Fresh pkg-config v0.3.27
       Fresh pin-project-lite v0.2.13
       Fresh crossbeam-utils v0.8.16
       Fresh opencv-binding-generator v0.77.0
       Fresh futures-macro v0.3.29
       Fresh regex-automata v0.4.3
       Fresh inotify-sys v0.1.5
       Fresh same-file v1.0.6
       Fresh bitflags v1.3.2
       Fresh crossbeam-channel v0.5.8
       Fresh mio v0.8.9
       Fresh num-traits v0.2.17
       Fresh is-terminal v0.4.9
       Fresh filetime v0.2.22
       Fresh termcolor v1.4.0
       Fresh futures-util v0.3.29
       Fresh inotify v0.9.6
       Fresh walkdir v2.4.0
       Fresh regex v1.10.2
       Fresh fastrand v2.0.1
       Fresh humantime v2.1.0
       Fresh futures-executor v0.3.29
       Fresh notify v6.1.1
       Fresh tempfile v3.8.1
       Fresh env_logger v0.10.1
       Fresh futures v0.3.29
       Fresh opencv v0.86.1
   Compiling focus_tool v0.1.0 (/develop/Code/focus_tool)
     Running `CARGO=/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/bin/cargo CARGO_BIN_NAME=focus_tool CARGO_CRATE_NAME=focus_tool CARGO_MANIFEST_DIR=/develop/Code/focus_tool CARGO_PKG_AUTHORS='' CARGO_PKG_DESCRIPTION='' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE='' CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=focus_tool CARGO_PKG_README=README CARGO_PKG_REPOSITORY='' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=0.1.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=1 CARGO_PKG_VERSION_PATCH=0 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 LD_LIBRARY_PATH='/develop/Code/focus_tool/target/debug/deps:/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib:/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib' /home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/bin/rustc --crate-name focus_tool --edition=2021 src/main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=155 --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 -C metadata=607ae46b30388a38 -C extra-filename=-607ae46b30388a38 --out-dir /develop/Code/focus_tool/target/debug/deps -C incremental=/develop/Code/focus_tool/target/debug/incremental -L dependency=/develop/Code/focus_tool/target/debug/deps --extern env_logger=/develop/Code/focus_tool/target/debug/deps/libenv_logger-1eb3a90c822f7164.rlib --extern futures=/develop/Code/focus_tool/target/debug/deps/libfutures-920f5c99ffce88fd.rlib --extern log=/develop/Code/focus_tool/target/debug/deps/liblog-f20c7c7a183f4e2a.rlib --extern notify=/develop/Code/focus_tool/target/debug/deps/libnotify-090ee37132e860e1.rlib --extern opencv=/develop/Code/focus_tool/target/debug/deps/libopencv-6d3b96e4f3c0ffc5.rlib --extern tempfile=/develop/Code/focus_tool/target/debug/deps/libtempfile-d4dcc9acb0518e5f.rlib -L native=/develop/Code/focus_tool/target/debug/build/opencv-5d009c3067acb7d6/out -L /usr/local/lib`
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/bin:/home/nvidia/.vscode-server/bin/2b35e1e6d88f1ce073683991d1eff5284a32690f/bin/remote-cli:/home/nvidia/.cargo/bin:/home/nvidia/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/nvidia/bin" VSLANG="1033" "cc" "/tmp/rustcrWyCpu/symbols.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.10j9vsuh8t7tsacw.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.11ffu2ubnhie36nb.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.14d7jhwp38ta55ir.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.195mtdy6q34g03fa.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1b4jxspwdzqbd8nm.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1dpizmlqcbt029el.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1jy8b3p7uyp9596x.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1m2vonrpsektqps0.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1osbcc6ovotw0qp3.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1pp4vgvz30dzb010.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1pswuwx1391yk2ev.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1s0sxuo71oftvnmq.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1t3v79dbw323zei1.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.1y622hdxss25f5rz.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.206pisdxkc89rkdq.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.20u7l6g7rdcwp4s1.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.276g0tn4unu3btp9.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2796kae9wr1gx7g5.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.27vq76g725dymxk9.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.28vwb8bkn0u9z06o.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2bhmjj9xlq04cibp.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2fk7224iczukzhbv.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2g3vvnduezwomvjc.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2hbbgkginvpypk2k.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2nvmq51ts41unru8.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2o80ruvzgo16lqxz.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2oi1ot5shfy3p3jt.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2r6lrtbw8q5egv02.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2visbxszyml8hdad.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.2xjw5ny8kygc7xdg.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.31ni0suwqfuej0mz.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.38mpzvhkgyrk3l2b.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.395df6ht10ekjjsl.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.39q6dq92a0u1spe3.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.3bjc3vgh4l1rfk01.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.3dsur9ojv4j9wbgg.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.3o7mb35vfyu9li6j.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.3yk1wnfkjphw9pl3.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.41349vkzgkf2cuyf.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.44fm0ow5rd5i6k3f.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.44nyktjrwjvgqhfp.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.455fmnvdqh5njr5q.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.46mnks4f43nw9edc.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.46ti5i3tmpyav6mj.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.47e5dd7h76dyrpaa.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.47k0d0jtujmqe9x1.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.48f7cwn40t0snkbo.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.4kd7tnowhqcqdhkh.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.4oigj4yd7bzznlm2.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.4oo7nbovr9d398c0.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.4r1y27it1cwk1830.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.4zckva3l0q3eyqps.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.4zxq87vpanzwg1ue.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.53p6qb57i1j1dtco.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.53yu9qnm82os70kc.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.59mrxq3xi9mwmykt.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.5a3q7mmevateqpa4.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.5ac9p5jdpn9tu59g.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.5gfx0j0fxi4jhm76.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.b8wtfbv197nxob5.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.cvva057vj3acm2g.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.ei83yawt77eppue.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.f5lxo1o4gfdjuvh.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.kasy12kis55s94b.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.lbk06cysesybgsh.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.pzyce1jfxsae4t9.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.ssdh4vdln390x5q.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.tv9uq3g1k5aabtu.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.v6nhyedjlcu7hlh.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.x0oqglq7ee6qbyw.rcgu.o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38.8kiowhzigbwmmfo.rcgu.o" "-Wl,--as-needed" "-L" "/develop/Code/focus_tool/target/debug/deps" "-L" "/develop/Code/focus_tool/target/debug/build/opencv-5d009c3067acb7d6/out" "-L" "/usr/local/lib" "-L" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/develop/Code/focus_tool/target/debug/deps/libnotify-090ee37132e860e1.rlib" "/develop/Code/focus_tool/target/debug/deps/libmio-18d857f938437127.rlib" "/develop/Code/focus_tool/target/debug/deps/libcrossbeam_channel-9bd1bd34d2656085.rlib" "/develop/Code/focus_tool/target/debug/deps/libcrossbeam_utils-911266bec6569e3a.rlib" "/develop/Code/focus_tool/target/debug/deps/liblog-f20c7c7a183f4e2a.rlib" "/develop/Code/focus_tool/target/debug/deps/libfiletime-c329aa101ea0c967.rlib" "/develop/Code/focus_tool/target/debug/deps/libcfg_if-e42da6bd2990ca07.rlib" "/develop/Code/focus_tool/target/debug/deps/libwalkdir-847b76d881bbdc6f.rlib" "/develop/Code/focus_tool/target/debug/deps/libsame_file-1f4e481a9c4c4e20.rlib" "/develop/Code/focus_tool/target/debug/deps/libinotify-905b24ab4c4de9dc.rlib" "/develop/Code/focus_tool/target/debug/deps/libinotify_sys-ace4324cc13ec4ed.rlib" "/develop/Code/focus_tool/target/debug/deps/libbitflags-379f7419f3592d3b.rlib" "/develop/Code/focus_tool/target/debug/deps/libfutures-920f5c99ffce88fd.rlib" "/develop/Code/focus_tool/target/debug/deps/libfutures_executor-ca401b81baa3ca80.rlib" "/develop/Code/focus_tool/target/debug/deps/libfutures_util-0e5cd6ba99b3f231.rlib" "/develop/Code/focus_tool/target/debug/deps/libmemchr-bfcd7c26cd24dc54.rlib" "/develop/Code/focus_tool/target/debug/deps/libfutures_io-fbd8d6c69b5b4db4.rlib" "/develop/Code/focus_tool/target/debug/deps/libslab-95ef085fa094e9fa.rlib" "/develop/Code/focus_tool/target/debug/deps/libfutures_channel-4bb86141ba4b8ed7.rlib" "/develop/Code/focus_tool/target/debug/deps/libpin_project_lite-bf94ea2abb38df53.rlib" "/develop/Code/focus_tool/target/debug/deps/libfutures_sink-6fda3165026d5c97.rlib" "/develop/Code/focus_tool/target/debug/deps/libfutures_task-90a2859df188640f.rlib" "/develop/Code/focus_tool/target/debug/deps/libpin_utils-e1896c8b0b5423aa.rlib" "/develop/Code/focus_tool/target/debug/deps/libfutures_core-0ee420a9ee991812.rlib" "/develop/Code/focus_tool/target/debug/deps/libopencv-6d3b96e4f3c0ffc5.rlib" "/develop/Code/focus_tool/target/debug/deps/liblibc-b86280da9908df62.rlib" "/develop/Code/focus_tool/target/debug/deps/libnum_traits-b274ff1cb84ae004.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libstd-dab4abc78347b511.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libpanic_unwind-28d6b1ab6bd269ff.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libobject-1815b83f04d33a2a.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libmemchr-0d260dd97de5e107.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libaddr2line-8fe33e9d81e1b59c.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libgimli-20dab92f1924d5b2.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/librustc_demangle-af1eaa2a70b1ba25.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libstd_detect-4771fea4fbb1028b.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libhashbrown-a2988562b6d2027d.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-40958581254a2d46.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libminiz_oxide-838c2816e28aa37c.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libadler-08330b78ec226d74.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libunwind-989d1e8eb1d72205.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcfg_if-1d3aa5179fd4d60d.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/liblibc-b69040e40940eb31.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/liballoc-8d6e9cd4e5af5f76.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/librustc_std_workspace_core-6d546367199a8518.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcore-2df04ccdcb2da4d8.rlib" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcompiler_builtins-e98df64cadd875cc.rlib" "-Wl,-Bdynamic" "-lstdc++" "-lopencv_calib3d" "-lopencv_imgproc" "-lopencv_contrib" "-lopencv_legacy" "-lopencv_core" "-lopencv_ml" "-lopencv_features2d" "-lopencv_objdetect" "-lopencv_flann" "-lopencv_video" "-lopencv_highgui" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-o" "/develop/Code/focus_tool/target/debug/deps/focus_tool-607ae46b30388a38" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: /usr/bin/ld: cannot find -lopencv_contrib
          /usr/bin/ld: cannot find -lopencv_legacy
          collect2: error: ld returned 1 exit status
          

error: could not compile `focus_tool` (bin "focus_tool") due to previous error

Caused by:
  process didn't exit successfully: `CARGO=/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/bin/cargo CARGO_BIN_NAME=focus_tool CARGO_CRATE_NAME=focus_tool CARGO_MANIFEST_DIR=/develop/Code/focus_tool CARGO_PKG_AUTHORS='' CARGO_PKG_DESCRIPTION='' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE='' CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=focus_tool CARGO_PKG_README=README CARGO_PKG_REPOSITORY='' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=0.1.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=1 CARGO_PKG_VERSION_PATCH=0 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 LD_LIBRARY_PATH='/develop/Code/focus_tool/target/debug/deps:/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib:/home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib' /home/nvidia/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/bin/rustc --crate-name focus_tool --edition=2021 src/main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=155 --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 -C metadata=607ae46b30388a38 -C extra-filename=-607ae46b30388a38 --out-dir /develop/Code/focus_tool/target/debug/deps -C incremental=/develop/Code/focus_tool/target/debug/incremental -L dependency=/develop/Code/focus_tool/target/debug/deps --extern env_logger=/develop/Code/focus_tool/target/debug/deps/libenv_logger-1eb3a90c822f7164.rlib --extern futures=/develop/Code/focus_tool/target/debug/deps/libfutures-920f5c99ffce88fd.rlib --extern log=/develop/Code/focus_tool/target/debug/deps/liblog-f20c7c7a183f4e2a.rlib --extern notify=/develop/Code/focus_tool/target/debug/deps/libnotify-090ee37132e860e1.rlib --extern opencv=/develop/Code/focus_tool/target/debug/deps/libopencv-6d3b96e4f3c0ffc5.rlib --extern tempfile=/develop/Code/focus_tool/target/debug/deps/libtempfile-d4dcc9acb0518e5f.rlib -L native=/develop/Code/focus_tool/target/debug/build/opencv-5d009c3067acb7d6/out -L /usr/local/lib` (exit status: 1)
   ```

**Comment by twistedfall** [comment]
The full build log as produced by `cargo build -vv` (after doing `cargo clean`) would help, but I suspect that the list of libraries to link is coming from the OpenCV discovered through pkg-config or cmake. And this list contains non-existent libraries for some reason. One way to fix this is to specify only the necessary libraries with the help of `OPENCV_LINK_LIBS` environment variable: https://github.com/twistedfall/opencv-rust#environment-variables

**Comment by buddhisthead** [comment]
@twistedfall Thank you so much! Yes, that worked great. I built with this command:
`OPENCV_LINK_LIBS="opencv_imgproc,opencv_imgcodecs,opencv_core" OPENCV_LINK_PATHS=/usr/lib/ `OPENCV_INCLUDE_PATHS=/usr/include/opencv4 cargo build

Following your hints has also solved the original problem. I had created my own pc file, because our install is done by installing from a gz file and doesn't use the package manager. The last revision of that included libraries that didn't exist because of my iterations on the project that wanted those extra libraries; and I forgot to remove them from the pc file as shown here. Note contrib, legacy and highgui are present (this is the error).
```
prefix=/usr
exec_prefix=${prefix}/local
includedir=${prefix}/include/opencv4
libdir=${exec_prefix}/lib

Name: opencv2
Description: The opencv library
Version: 4.5.5
Cflags: -I${includedir}
Libs: -L${libdir} -lopencv_calib3d -lopencv_imgproc -lopencv_contrib -lopencv_legacy -lopencv_core -lopencv_ml -lopencv_features2d -lopencv_objdetect -lopencv_flann -lopencv_video -lopencv_highgui
```

And in my .bashrc I have:
```
PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/lib/pkgconfig
export PKG_CONFIG_PATH
```

After pairing the pkg config file down to the actual list of existing libraries, it builds with `cargo build`. Thank you for the amazing support and the awesome rust crate! Let me know if you make it to Portland, OR and dinner is on me. Cheers.
